helix-view/src/editor.rs

```diff
@@ -73,6 +73,14 @@ where
     )
 }
 
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum DiffSource {
+    #[default]
```

Comment:
why did you remove the `None` vraint here?

I think its quite reasonable to allow people to turn the diff gutter off. 
Currently hiding the statusbar element will not actually disable the background diff computation but that may be desirable for better perf/battery life (especially if you have some random strang large file where you just don't case for the diff). Its a bit niche but I would be fine with adding it.